src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -211,6 +214,24 @@ public void transform(KeyValue<LongWritable, GenericData.Record> input, Emitter<
 
   @Override
   public void onRunFinish(boolean succeeded, BatchSourceContext context) {
+    deleteGcsTemporaryDirectory();
+    deleteBigQueryTemporaryTable();
+  }
+
+  private void deleteBigQueryTemporaryTable() {
+    String temporaryTable = configuration.get(BigQueryConstants.CONFIG_TEMPORARY_TABLE_NAME);
+    String serviceAccount = config.getServiceAccount();
+    try {
+      Credentials credentials = getCredentials(serviceAccount);
```

Comment:
handle null credentials here

Reply:
null credentials can also be as a result of auto-detect.

In `getCredentials` method we are already handling null for `serviceAccount`:

```
@Nullable
  private Credentials getCredentials(@Nullable String serviceAccount) throws IOException {
    return serviceAccount == null ?
            null : GCPUtils.loadServiceAccountCredentials(serviceAccount, config.isServiceAccountFilePath());
  }
```

And credentials null are also handled in `GCPUtils`, `getBigQuery`:

```
public static BigQuery getBigQuery(String project, @Nullable Credentials credentials) {
    BigQueryOptions.Builder bigqueryBuilder = BigQueryOptions.newBuilder().setProjectId(project);
    if (credentials != null) {
      bigqueryBuilder.setCredentials(credentials);
    }
    return bigqueryBuilder.build().getService();
  }
```

Reply:
I see